src/XMakeCommandLine/CommandLineSwitches.cs

```diff
@@ -438,14 +438,22 @@ bool unquoteParameters
             bool parametersStored = false;
 
             // if this is the first time this switch has been detected
-            if (_parameterizedSwitches[(int)parameterizedSwitch].commandLineArg == null)
+            if (_parameterizedSwitches[(int) parameterizedSwitch].commandLineArg == null)
             {
                 // initialize its parameter storage
-                _parameterizedSwitches[(int)parameterizedSwitch].parameters = new ArrayList();
-            }
+                _parameterizedSwitches[(int) parameterizedSwitch].parameters = new ArrayList();
 
-            // save the switch text
-            _parameterizedSwitches[(int)parameterizedSwitch].commandLineArg = commandLineArg;
+                // save the switch text
+                _parameterizedSwitches[(int) parameterizedSwitch].commandLineArg = commandLineArg;
+            }
+            else
+            {
+                _parameterizedSwitches[(int) parameterizedSwitch].commandLineArg = string.Concat(
+                    _parameterizedSwitches[(int)parameterizedSwitch].commandLineArg,
+                    " ",
+                    commandLineArg
+                    );
```

Comment:
Ah, but that's kind of explicit, since the command line is written with GetEquivalentCommandLineExceptProjectFile, featuring this comment which I find inscrutable:
```
Get the equivalent command line, with response files expanded
and duplicates removed. Prettified, sorted, parameterless first.
Don't include the project file, the caller can put it last.
```

So this is no worse than before.